In [1]:
from tensorflow.keras.preprocessing.text import one_hot

In [9]:
sent=[  'the glass of milk',
     'the glass of juice',
     'the cup of tea',
    'I am a good boy',
     'I am a good developer',
     'understand the meaning of words',
     'your videos are good',]

In [10]:
sent

['the glass of milk',
 'the glass of juice',
 'the cup of tea',
 'I am a good boy',
 'I am a good developer',
 'understand the meaning of words',
 'your videos are good']

In [11]:
# Define the vocabulary size
voc_size=10000

In [12]:
# One hot encoding representation
one_hot_rep=[one_hot(words,voc_size)for words in sent]

In [13]:
one_hot_rep[0]

[9942, 7942, 5064, 9084]

**the glass of milk** this vector mean that for "the"  the index 9942 it will be 1 and  all things will be 0

In [14]:
# word embedding represnetation

from tensorflow.keras.layers import Embedding
from tensorflow.keras.preprocessing.sequence import pad_sequences
from tensorflow.keras.models import Sequential

In [15]:
import numpy as np

In [16]:
sent_length=8
embedded_docs=pad_sequences(one_hot_rep,padding='pre',maxlen=sent_length)
print(embedded_docs)

[[   0    0    0    0 9942 7942 5064 9084]
 [   0    0    0    0 9942 7942 5064 1208]
 [   0    0    0    0 9942 8755 5064 8837]
 [   0    0    0  616 6674 4977 7093 1580]
 [   0    0    0  616 6674 4977 7093 8318]
 [   0    0    0 8932 9942 7513 5064 6995]
 [   0    0    0    0 2710  300  842 7093]]


In [17]:
# feature representation
dim = 10

In [20]:
model = Sequential()
model.add(Embedding(voc_size,dim,input_length=sent_length))
model.compile('adam','mse')

/usr/local/lib/python3.11/dist-packages/keras/src/layers/core/embedding.py:90: UserWarning: Argument `input_length` is deprecated. Just remove it.
  warnings.warn(


In [21]:
model.summary()

Model: "sequential_1"

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ Layer (type)                         ┃ Output Shape                ┃         Param # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ embedding_1 (Embedding)              │ ?                           │     0 (unbuilt) │
└──────────────────────────────────────┴─────────────────────────────┴─────────────────┘

 Total params: 0 (0.00 B)

 Trainable params: 0 (0.00 B)

 Non-trainable params: 0 (0.00 B)

In [22]:
model.predict(embedded_docs)[0]

1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 290ms/step


array([[ 0.04385901,  0.03053769, -0.01768018, -0.00370878, -0.04265765,
         0.00543185,  0.00646304,  0.01714113,  0.02772817,  0.04168758],
       [ 0.04385901,  0.03053769, -0.01768018, -0.00370878, -0.04265765,
         0.00543185,  0.00646304,  0.01714113,  0.02772817,  0.04168758],
       [ 0.04385901,  0.03053769, -0.01768018, -0.00370878, -0.04265765,
         0.00543185,  0.00646304,  0.01714113,  0.02772817,  0.04168758],
       [ 0.04385901,  0.03053769, -0.01768018, -0.00370878, -0.04265765,
         0.00543185,  0.00646304,  0.01714113,  0.02772817,  0.04168758],
       [ 0.00839146,  0.0300534 ,  0.03869182,  0.03882593,  0.01761922,
         0.04788161, -0.01956167,  0.01471991, -0.03114849, -0.02116417],
       [-0.00592683, -0.00544529, -0.01004372,  0.02863452,  0.00200053,
         0.02460451,  0.0437923 , -0.03361678,  0.00818674,  0.02923018],
       [ 0.01800016,  0.02729538,  0.00178925,  0.0199831 , -0.02535957,
        -0.01902232, -0.04075072, -0.00398252

In [23]:
embedded_docs[0]

array([   0,    0,    0,    0, 9942, 7942, 5064, 9084], dtype=int32)